README.md

```diff
@@ -31,17 +31,43 @@ about the weird parts of working in a long-lived app.
 
     ember install ember-lifeline
 
-To use any of the below mentioned methods in your component, route or service, you will have to import and apply one or both of these mixins to your class:
-* `ember-lifeline/mixins/run` for using any of the *Task methods
-* `ember-lifeline/mixins/dom` for using `addEventListener`
-* `ember-lifeline/mixins/disposable` for using `registerDisposable` and `runDisposable`
-
 ## Usage
 
+Ember Lifeline supports a functional API that enables entanglement - _the association of async behavior to object instances_. This allows you to write async code in your classes that can be automatically cleaned up for you when the object is destroyed.
+
+The API is divided into two main parts:
+
+- Run loop entanglement
+- DOM event handler entanglement
+
+Additionally, lifeline exposes a primative, `disposables`, that allows you to entangle functionality of your choosing.
+
+:warning: When importing and using lifeline's functions, it's imperative that you additionally import and call `runDisposables` during your object's `destroy` method. This ensures lifeline will correctly dispose of any remaining async work.
+
+```
+import Component from '@ember/component';
+import { runTask, runDisposables } from 'ember-lifeline';
+
+export default Component.extend({
+  // use `runTask` method somewhere in this component
+
+  destroy() {
+    runDisposables(this); // ensure that lifeline will clean up any remaining async work
+
+    this._super(...arguments);
+  }
+})
+```
+
+When using the mixin, this is implemented for you.
```

Comment:
Just skimming this, but I’m not sure if this section is _above_ the mixin docs section. If it is, it’s odd to mention “the mixin” before defining it...